specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/webapplicationfirewall.json

```diff
@@ -0,0 +1,575 @@
+        },
+        "action": {
+          "description": "Type of Actions",
+          "type": "string",
+          "enum": [
+            "Allow",
+            "Block",
+            "Log"
+          ],
+          "x-ms-enum": {
+            "name": "action",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "Transform": {
+      "description": "Describes what transforms applied before matching",
+      "type": "string",
+      "enum": [
+        "Lowercase",
+        "Trim",
+        "UrlDecode",
+        "UrlEncode",
+        "RemoveNulls",
+        "HtmlEntityDecode"
+      ],
+      "x-ms-enum": {
+        "name": "transform",
+        "modelAsString": true
+      }
+    },
+    "MatchVariable": {
+      "description": "Define match variables",
+      "required": [  
+        "name"
+      ],
+      "properties": {
+        "name": {
+          "description": "Match Variable",
+          "type": "string",
+          "enum": [
+            "RemoteAddr",
+            "RequestMethod",
+            "QueryString",
+            "PostArgs",
+            "RequestUri",
+            "RequestHeaders",
+            "RequestBody",
+            "RequestCookies"
+          ],
+          "x-ms-enum": {
+            "name": "matchVariable",
+            "modelAsString": true
+          }
+        },
+        "selector": {
+          "description": "Describes field of the matchVariable collection",
+          "type": "string"
+        }
+      }
+    },
+    "MatchCondition": {
+      "description": "Define match conditions",
+      "required": [  
+        "matchVariables",  
+        "operator", 
+        "matchValues"
+      ],
+      "properties": {
+        "matchVariables": {
+          "description": "List of match variables",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/MatchVariable"
+          }
+        },
+        "operator": {
+          "description": "Describes operator to be matched",
+          "type": "string",
+          "enum": [
+            "IPMatch",
+            "Equal",
+            "Contains",
+            "LessThan",
+            "GreaterThan",
+            "LessThanOrEqual",
+            "GreaterThanOrEqual",
+            "BeginsWith",
+            "EndsWith",
+            "Regex"
+          ],
+          "x-ms-enum": {
+            "name": "operator",
+            "modelAsString": true
+          }
+        },
+        "negateCondition": {
+          "description": "Describes if this is negate condition or not",
+          "type": "boolean"
+        },
+        "matchValues": {
+          "description": "Match value",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "transforms": {
+          "description": "List of transforms",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Transform"
+          }
+        }
+      }
+    },
+    "ErrorResponse": {
```

Comment:
NetworkWatcher's specs already have definition named `ErrorResponse`. I think you can use `ErrorDetails` from network.json here.
@jianghaolu please confirm

Reply:
Yes please re-use if it's the same model. Thanks @number213

Reply:
@jianghaolu Its not exactly the same. So, skipping.